src/main/java/uk/ac/cam/cl/dtg/segue/api/EmailFacade.java

```diff
@@ -298,6 +301,79 @@ public Response generateEmailVerificationToken(@Context final HttpServletRequest
+     * @param request - to get the current user from.
+     * @return a 204 No Content on success, or an error.
+     */
+    @POST
+    @Path("/users/deleteaccount")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Initiate an account deletion request.",
+            description = "If the account email is not invalid, this sends a deletion token by email.")
+    public Response requestAccountDeletion(@Context final HttpServletRequest request) {
+
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
+
+            if (EmailVerificationStatus.DELIVERY_FAILED.equals(currentUser.getEmailVerificationStatus())) {
+                return SegueErrorResponse.getBadRequestResponse("Only accounts with valid email addresses can request deletion using this method.");
+            }
+
+            if (!Role.STUDENT.equals(currentUser.getRole())) {
+                return new SegueErrorResponse(Status.FORBIDDEN, "Only student accounts can request deletion using this method").toResponse();
+            }
+
+            userManager.sendAccountDeletionEmail(currentUser);
+
+            this.getLogManager().logEvent(currentUser, request, SegueServerLogType.ACCOUNT_DELETION_REQUEST_RECEIVED, Maps.newHashMap());
+
+            return Response.noContent().build();
+
+        } catch (NoUserLoggedInException e) {
+            return SegueErrorResponse.getNotLoggedInResponse();
+        } catch (SegueDatabaseException | ContentManagerException e) {
+            log.warn("Failed to create and send an account deletion email!", e);
+            return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, "Error sending account deletion message.", e).toResponse();
+        }
+    }
+
+    /**
+     * Endpoint to delete an account using an account deletion token.
+     *
+     * @param request - to get the current user from.
+     * @param token - the deletion token previously sent to the account email.
+     * @return a 204 No Content on success, or an error.
+     */
+    @DELETE
+    @Path("/users/deleteaccount")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Complete an account deletion request using a deletion token.",
+               description = "The user must be logged in to the account to delete to complete this action.")
+    public Response completeAccountDeletion(@Context final HttpServletRequest request,
+                                            @Context final HttpServletResponse response,
+                                            @QueryParam("token") final String token) {
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
+
+            userManager.confirmDeletionTokenAndDeleteAccount(currentUser, token);
+
+            userManager.logUserOut(request, response);
+
+            this.getLogManager().logEvent(currentUser, request, SegueServerLogType.ACCOUNT_DELETION_REQUEST_COMPLETE, Maps.newHashMap());
```

Comment:
I don't believe `logUserOut` can fail; it doesn't do anything that can throw an exception except invalidate the session, and that is caught by the function. This is also consistent with every other log event, where indeed since we don't do transactions in many places, things can partially succeed and/or not be logged. 
I think it is as likely to fail inside the deletion after the transaction (one place we do do transactions!) as it is to fail here!